static/app/views/eventsV2/table/columnEditModal.spec.jsx

```diff
@@ -20,13 +24,45 @@ function mountModal({columns, onApply, customMeasurements}, initialData) {
       closeModal={() => void 0}
       customMeasurements={customMeasurements}
     />,
-    initialData.routerContext
+    {context: initialData.routerContext}
   );
 }
 
-describe('EventsV2 -> ColumnEditModal', function () {
-  enforceActOnUseLegacyStoreHook();
+const findAllQueryFields = () => screen.findAllByTestId('queryField');
+const findAllQueryFieldNthCell = async nth =>
+  (await findAllQueryFields())
+    .map(f => within(f).getAllByTestId('label')[nth])
+    .filter(Boolean);
+const getAllQueryFields = () => screen.getAllByTestId('queryField');
+const getAllQueryFieldsNthCell = nth =>
+  getAllQueryFields()
+    .map(f => within(f).getAllByTestId('label')[nth])
+    .filter(Boolean);
+
+const openMenu = async (row, column = 0) => {
+  const queryFields = await screen.findAllByTestId('queryField');
+  const queryField = queryFields[row];
+  expect(queryField).toBeInTheDocument();
+
+  const labels = within(queryField).queryAllByTestId('label');
+  if (labels.length > 0) {
+    userEvent.click(labels[column]);
+  } else {
+    // For test adding a new column, no existing label.
+    const placeholder = screen.getByText('(Required)');
+    expect(placeholder).toBeInTheDocument();
+    userEvent.click(placeholder);
+  }
+};
+
+const selectByLabel = async (label, options) => {
+  await openMenu(options.at);
+  const menuOptions = await screen.findAllByTestId('menu-list-item-label');
```

Comment:
does this menu item has a role we could use together with `screen.getByRole`?

Reply:
There is a parent `menuitem` but we want the label within the menu-item since otherwise matching to the textContent includes the 'f(x)' etc. from the badge, will leave a comment.